packages/core-plugin-api/src/plugin/Plugin.tsx

```diff
@@ -58,6 +60,19 @@ export class PluginImpl<
     return extension.expose(this);
   }
 
+  reconfigure(options: PluginInputOptions): void {
+    if (this.config.configure) {
+      this.config.options = this.config.configure(options);
+    }
+  }
+
+  getPluginOptions(): PluginOptions {
```

Comment:
Let's make this private, and then have the provider do a feature detection on the plugin, so basically casting it like this: `plugin as unknown as { getPluginOptions?(): PluginOptions }`